packages/docusaurus-theme-common/src/utils/blogUtils.ts

```diff
@@ -30,3 +30,22 @@ export function useVisibleBlogSidebarItems(
     [items, pathname],
   );
 }
+
+export function formatBlogPostDate(
```

Comment:
I'm not sure keeping this method as-is is our best option.

What about adding a new useful generic hooks in a new `IntlUtils.ts` file?

```ts
export function useCalendar(): string {
  const {
    i18n: {currentLocale, localeConfigs},
  } = useDocusaurusContext();
  return localeConfigs[currentLocale]!.calendar;
}

export function useDateTimeFormat(
  options: Intl.DateTimeFormatOptions = {},
): Intl.DateTimeFormat {
  const {
    i18n: {currentLocale},
  } = useDocusaurusContext();
  const calendar = useCalendar();
  return new Intl.DateTimeFormat(currentLocale, {
    calendar,
    ...options,
  });
}
```

This way you could directly call it in React components:


```tsx
function Year({year, posts}: YearProp) {
  const dateTimeFormat = useDateTimeFormat({
      day: 'numeric',
      month: 'long',
      year: undefined, // Can be customized just for this page
      timeZone: 'UTC'
  })

  const formattedDate = dateTimeFormat.format(post.metadata.date);
```

And we would be able to use it in other places later (not just blog posts but also docs components)